src/MongoDB.Driver.Core/Core/Clusters/ServerSelectors/CompositeServerSelector.cs

```diff
@@ -47,6 +47,10 @@ public IEnumerable<ServerDescription> SelectServers(ClusterDescription cluster,
             foreach (var selector in _selectors)
             {
                 selectedServers = selector.SelectServers(cluster, selectedServers);
+                if (!selectedServers.Any())
+                {
+                    break;
+                }
```

Comment:
Remove this code.

In the C# driver we call server selectors even when the list is empty.